### ISSUE TITLE
Unable to to set automountServiceAccountToken in podTemplate

### ISSUE THREAD

**Issue by codezomb** [elastic/cloud-on-k8s#1151]
## Bug Report
**What did you do?**

Tried to enable `automountServiceAccountToken`

```yaml
podTemplate:
      spec:
        automountServiceAccountToken: true
```

**What did you expect to see?**

https://github.com/elastic/cloud-on-k8s/blob/b6100d7d6ad8e181556b3e1b0586bd296bc74e6c/operators/pkg/controller/common/defaults/pod_template.go#L53-L55

based on this bit of code, I expected to be able to set `automountServiceAccountToken`

**What did you see instead? Under which circumstances?**

`automountServiceAccountToken` is not changed

**Environment**

* Version information:

0.8.1

* Kubernetes information:

Kops installed cluster v1.13

```
$ kubectl version
```
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:53:57Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

* Resource definition:
```
if relevant insert the resource definition
```

```yaml
apiVersion: elasticsearch.k8s.elastic.co/v1alpha1
kind: Elasticsearch
metadata:
  name: test-cluster
spec:
  version: 7.1.0
  nodes:
  - nodeCount: 1
    config:
      node.ingest: true
      node.master: true
      node.data: true
    podTemplate:
      spec:
        automountServiceAccountToken: true
```

**Comment by sebgl** [comment]
Hey @codezomb, thanks for reporting.

Unfortunately, this does indeed not work in release 0.8. Inheriting pod template is only [limited to a few fields (resources, labels and annotations)](https://github.com/elastic/cloud-on-k8s/blob/f9d2b799c05f55988c203572eabe3b9fbc47943b/operators/pkg/apis/elasticsearch/v1alpha1/elasticsearch_types.go#L68). I don't see a workaround you could use in the meantime, sorry :/

This is fixed on master, and will be fixed on upcoming release 0.9 (no precise release date yet, but probably before the end of the summer).

**Comment by sebgl** [comment]
If that's ok with you @codezomb I'll close this issue now since this is fixed already (just not released yet).
